### PR TITLE
Update how dataloader spans are handled

### PR DESCRIPTION
**Pull request by maciej-szlosarczyk**
* Add more attributes to batches
* Use deeper ancestor stack for processes that are being dataloaded to avoid orphan spans.

Signed-off-by: Maciej Szlosarczyk <maciej@mindvalley.com>

### CODE REVIEW

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **maciej-szlosarczyk** after the PR has been reviewed.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fmv-opentelemetry).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/mv-opentelemetry/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: lib/mv_opentelemetry/ecto.ex [hunk 1]
```diff
@@ -73,7 +73,7 @@ defmodule MvOpentelemetry.Ecto do
     span_name = name(config, event, source)
     span_opts = %{start_time: start_time, attributes: all_attributes, kind: :client}
 
-    parent_context = OpentelemetryProcessPropagator.fetch_parent_ctx(1, :"$callers")
+    parent_context = OpentelemetryProcessPropagator.fetch_parent_ctx(4, :"$callers")
```
Comment (maciej-szlosarczyk) [inline lib/mv_opentelemetry/ecto.ex:76]: Screenshot 2022-04-11 at 12 21 31

This is the best I can do...

File: lib/mv_opentelemetry/dataloader.ex [hunk 2]
```diff
@@ -27,23 +28,55 @@ defmodule MvOpentelemetry.Dataloader do
...
+        %{batch: batch},
+        opts
+      ) do
     event_name = "dataloader.source.batch.run"
 
-    parent_context = OpentelemetryProcessPropagator.fetch_parent_ctx(1, :"$callers")
-    attach_context(parent_context)
+    batch_attributes =
+      case batch do
+        {{:queryable, _pid, queryable, cardinality, col, _}, _} ->
```
Comment (maciej-szlosarczyk) [inline lib/mv_opentelemetry/dataloader.ex:41]: Some more attributes for DL, not sure how useful these are, but maybe we can do something with them.

File: lib/mv_opentelemetry/dataloader.ex [hunk 3]
```diff
@@ -27,23 +28,55 @@ defmodule MvOpentelemetry.Dataloader do
...
-  end
+        _ ->
+          []
+      end
+
+    %{duration: duration} = measurements
+    end_time = :opentelemetry.timestamp()
+    start_time = end_time - duration
+    attributes = batch_attributes ++ opts[:default_attributes]
+    span_opts = %{start_time: start_time, attributes: attributes}
```
Comment (maciej-szlosarczyk) [inline lib/mv_opentelemetry/dataloader.ex:65]: Like with ecto we need to cheat here, dataloader is not actually emitting span events, just an event at the beginning and at the end.

**Comment by sauron-droid** [comment]
LGTM label has been added.  <details>Git tree hash: 04ad0a0da3e469f68c233922f5b4a3b641a03af4</details>

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/mv-opentelemetry/pull/34#pullrequestreview-937721287" title="Approved">antonsatin</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fmv-opentelemetry).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/mindvalley/mv-opentelemetry/blob/main/OWNERS)~~ [antonsatin]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**